ethcore/light/src/client/service.rs

```diff
@@ -58,15 +60,15 @@ impl fmt::Display for Error {
 }
 
 /// Light client service.
-pub struct Service<T> {
+pub struct Service<T: 'static> {
 	client: Arc<Client<T>>,
-	io_service: IoService<ClientIoMessage>,
+	io_service: IoService<ClientIoMessage<()>>,
```

Comment:
why is this `()`?

does it work because we only care about `ClientIoMessage::BlockVerified` for the light client?

Reply:
Great question. So before this PR `light` depends on `ethcore` essentially only for `ClientIoMessage`, and then only because it has a `Execute` variant that embeds a `Callback` which is a `Fn` that takes an `ethcore::Client` as a param… So the dependency was actually not "real" but merely ceremonial, to satisfy the types as they were set up.
Once I made the `Callback` generic over the Client (and all the other types using it) @ordian spotted that there's actually not necessary to use anything other than `()` in `light`.

Now, we all agree that `ClientIoMessage<()>` looks plenty odd and hopefully someone will come along and refactor this further. ;)
In particular I know that @ngotchac is working on changes to the IO code that may, as a desired side-effect, result in a better message passing infrastructure, one that does not need a `Client` (or whatever stand-in we use).

Reply:
Ok, I see.
> In particular I know that @ngotchac is working on changes to the IO code that may, as a desired side-effect, result in a better message passing infrastructure, one that does not need a Client (or whatever stand-in we use).

Cool :+1: